### PR TITLE
Modal body scroll & overflow refactor

### PR DESCRIPTION
**Pull request by bcfancher-mc**
## Overview
Refactoring Modal component to simplify the containers, make them a little more legible, and to clean up the body anti-scrolling and long, scrolling modals.

## Risks
Low - impacts MOST modals in mc (there are some plain-old bootstrap modals still)

## Changes
no intended behavioral differences, except contents are no longer cutoff

## Issue
https://github.com/yankaindustries/mc-components/issues/628

## Breaking change?
backwards compatible

### CODE REVIEW

File: src/components/Modal/index.js [hunk 1]
```diff
@@ -50,18 +50,18 @@ export default class Modal extends PureComponent {
...
     const { show } = this.props
-
-    // a little hackery here to prevent scrolling/reset
+    const rootHtml = document.getElementsByTagName('html')[0]
 
     if (!prevProps.show && show) {
-      const scrollOffset = window.scrollY
-      document.body.style.top = `-${scrollOffset}px`
       document.body.classList.add('mc-modal__body--open')
+      rootHtml.classList.add('mc-modal__html--open')
```
Comment (stephenp) [inline src/components/Modal/index.js]: This here should only need the root.  From there, in the css we can target it like this if we need both overflows:

```
.mc-modal__body--open {
  overflow: hidden;
  body { overflow: hidden; }
}
```

File: src/components/Modal/index.stories.js [hunk 2]
```diff
@@ -44,8 +45,8 @@ class ModalExample extends Component {
         />
 
         <DocSection title='Demo'>
-          <div className='row'>
-            <div className='col-md-3'>
+          <div className='row' style={{ height: '2000px' }}>
```
Comment (stephenp) [inline src/components/Modal/index.stories.js]: I'm guessing this is so the page itself forces a scrollbar?  If so, can you move this to a standalone div after the `.row`?

Comment (stephenp) [reply src/components/Modal/index.stories.js]: (Don't want people confused as to why it needs an inline style on the row if they inspect it)

File: src/styles/components/_modal.scss [hunk 3]
```diff
@@ -32,38 +32,29 @@
...
+  &__viewport {
+    position: relative;
+    width: auto;
+    margin: 1.75rem 0;
+    z-index: $mc-zindex-modal-backdrop;
+
+    // vertically centered stuffs
+    display: flex;
+    align-items: center;
+    min-height: 95vh;
```
Comment (stephenp) [inline src/styles/components/_modal.scss]: Should this just be "height"?

File: src/components/Modal/index.js (hunk 1 above)
Comment (bcfancher-mc) [reply src/components/Modal/index.js]: good catch

File: src/components/Modal/index.stories.js (hunk 2 above)
Comment (bcfancher-mc) [reply src/components/Modal/index.stories.js]: good point

File: src/styles/components/_modal.scss (hunk 3 above)
Comment (bcfancher-mc) [reply src/styles/components/_modal.scss]: nope, has to be min-heigh, apparently

**Comment by jstnjns** [comment]
love it 😍

**Comment by jstnjns** [comment]
before we merge, can we do a sanity check with mc? link locally and check major usages?

**Comment by bcfancher-mc** [comment]
> before we merge, can we do a sanity check with mc? link locally and check major usages?

I did this yesterday

**Comment by jstnjns** [comment]
of course you did you wonderful man

**Comment by jstnjns** [comment]
fill out the PR description for the records, but the code looks g2g!

File: src/components/Modal/index.js [hunk 4]
```diff
@@ -50,18 +50,19 @@ export default class Modal extends PureComponent {
...
     } else if (prevProps.show && !show) {
-      const scrollOffset = -parseInt(document.body.style.top, 10)
-      document.body.classList.remove('mc-modal__body--open')
-      document.body.style.top = undefined
-      window.scrollTo(0, scrollOffset)
+      rootHtml.classList.remove('mc-modal__html--open')
+    }
+
+    if (show) {
+      const elem = this.container.current
```
Comment (stephenp) [inline src/components/Modal/index.js]: Thanks stack overflow!

Comment (bcfancher-mc) [reply src/components/Modal/index.js]: si senor!

**Pull request merged**